docs/plots/backends.rst

```diff
@@ -0,0 +1,389 @@
+************************
+Plotting backend details
+************************
+
+.. warning::
+   The plotting backend API is currently re-designed and should currently be considered
+   experimental.
+
+Plotting in Sherpa is done through *plotting backends*.
+The term *backend* is used in two related ways in Sherpa. First, a *backend* is an
+external package that performs the plotting such as :term:`matplotlib`, i.e. it creates a
+canvas and puts color on paper or on the screen. The Sherpa plotting classes are
+written in a backend-independent way; they work with any plotting backend that
+is supported by Sherpa. We also use the term *backend* to refer to Sherpa
+classes that connect the Sherpa plot objects with the plotting backend by
+translating Sherpa options into backend-specific commands. The details of that
+are explained below, but, in short, a backend class would translate the sherpa
+backend-independent "plot a line in red from here to there" into backend
+specific ``plt.plot(x,y, linecolor='r')`` (if the backend is :term:`matplotlib`).
+
+Instead of utilizing Sherpa classes and commands to plot data, users can
+alternatively just access the data in the Sherpa data objects (e.g. the x and y
+values of a dataset) and perform plotting operations directly with any plotting
+backend that is available to them. This method may be less convenient, but it
+works for plotting backends not (yet) supported by Sherpa.
+
+Which backend is used?
+======================
+
+When the `sherpa.plot` module is first imported, Sherpa tries to import the
+backends installed with Sherpa in the order listed in the
+``options.plot_pkg`` setting from the ``sherpa.rc`` startup file.
+The first module that imports successfully is set as the active
+backend. The following command prints the name and the location
+on disk of that module::
+
+   >>> from sherpa import plot
+   >>> print(plot.backend.name)
+
+.. warning::
+    Of course, you could be tempted to write::
+
+      >>> from sherpa.plot import backend
+      >>> print(backend.name)
+
+    However, ``backend`` is now a reference to the backend that
+    was active when the import was done. If the backend is changed
+    later, ``backend`` will still refer to the old backend. Thus,
+    always use ``plot.backend`` to access the active backend.
+
+Change the backend
+------------------
+
+.. warning::
+   When the plotting backend is changed, new plotting objects must be
+   initialized because objects like e.g. `sherpa.plot.DataPlot` keep a
+   copy of all default parameters of the backend that was active when they
+   are initialized. A different backend might not understand all those
+   parameters.
+
+   In particular, Sherpa's UI layer (which contains functions such as
+   `sherpa.astro.ui.plot_data`) keeps a reference to specific plot objects.
+   Changing the backend by itself will not change those references, and thus
+   those function might not work with a new backend.
```

Comment:
If they've used the `set-plot_backend` from the `Session` class then this should only be a problem for any objects stored by the user (e.g. after `get_data_plot`), as - I hope - `plot_data` will pick up the newly-created object.

But I haven't yet tested that.